db/db.go

```diff
@@ -189,14 +190,35 @@ func addClientCertFromBytes(cfg *tls.Config, data []byte, keyPasswd string) (str
 			certDecodedBlock = currentBlock.Bytes
 			start += len(certBlock)
 		} else if strings.HasSuffix(currentBlock.Type, "PRIVATE KEY") {
-			if keyPasswd != "" && x509.IsEncryptedPEMBlock(currentBlock) {
-				var encoded bytes.Buffer
-				buf, err := x509.DecryptPEMBlock(currentBlock, []byte(keyPasswd))
-				if err != nil {
-					return "", err
+			isEncrypted := x509.IsEncryptedPEMBlock(currentBlock) || strings.Contains(currentBlock.Type, "ENCRYPTED")
+			if isEncrypted {
+				if keyPasswd == "" {
+					return "", fmt.Errorf("no password provided to decrypt private key")
+				}
+
+				var keyBytes []byte
+				var err error
+				// Process the X.509-encrypted or PKCS-encrypted PEM block.
+				if x509.IsEncryptedPEMBlock(currentBlock) {
+					// Only covers encrypted PEM data with a DEK-Info header.
+					keyBytes, err = x509.DecryptPEMBlock(currentBlock, []byte(keyPasswd))
+					if err != nil {
+						return "", err
+					}
+				} else if strings.Contains(currentBlock.Type, "ENCRYPTED") {
+					// The pkcs8 package only handles the PKCS #5 v2.0 scheme.
+					decrypted, err := pkcs8.ParsePKCS8PrivateKey(currentBlock.Bytes, []byte(keyPasswd))
+					if err != nil {
+						return "", err
+					}
```

Comment:
From my understanding, there's a couple ways to encrypt a PEM key in PKCS 8 format. The only one that the `pkcs8` Go package can currently handle is called the PKCS 5 v2.0 scheme. The first time I tried this, I accidentally used the wrong scheme and got this error:

`error configuring the connector: error configuring client, can't load client certificate: pkcs8: only PKCS #5 v2.0 supported`

Even though the error is probably specific enough, I'm wondering if we should make a note of this in the docs under the [`--sslPEMKeyFile` entry](https://docs.mongodb.com/database-tools/mongodump/#cmdoption-mongodump-sslpemkeyfile) and provide guidance on getting the right scheme (e.g. using openssl with the `-v2 des3` option to get PKCS 5 v2.0).

I'm okay either way, but curious to hear others' thoughts.

Reply:
Nice catch! I agree we should add this limitation to the doc so that it won't cause confusion to the client.

Reply:
I'm not sure that this is a limitation significant enough to warrant a mention in the docs. Generally, the docs describe what we _do_ support, and I think a limitation has to be pretty notable/important for us to want to describe what we _don't_ do.